src/screens/Wallet.js

```diff
@@ -75,61 +73,108 @@ class Wallet extends React.Component {
     successMessage: '',
     hasTokenSignature: false,
     shouldShowAdministrativeTab: false,
+    totalSupply: null,
+    canMint: false,
+    canMelt: false,
+    transactionsCount: null,
+    mintCount: null,
+    meltCount: null,
   };
 
-  // Reference for the TokenGeneralInfo component
-  generalInfoRef = React.createRef();
-
-  // Reference for the TokenAdministrative component
-  administrativeRef = React.createRef();
-
   // Reference for the unregister confirm modal
   unregisterModalRef = React.createRef();
 
-  componentDidMount = () => {
+  componentDidMount = async () => {
     this.setState({
       backupDone: hathorLib.wallet.isBackupDone()
     });
 
-    $('#wallet-div').on('show.bs.tab', 'a[data-toggle="tab"]', (e) => {
-      // On tab show we reload token data from the server for each component
-      if (e.target.id === 'token-tab') {
-        this.generalInfoRef.current.updateTokenInfo();
-      } else if (e.target.id === 'administrative-tab') {
-        this.administrativeRef.current.updateTokenInfo();
-      }
-    });
+    await this.updateTokenInfo(this.props.selectedToken);
 
+    this.updateWalletInfo();
     // First time the screen is mounted we must also check if we should show administrative tab
     this.shouldShowAdministrativeTab(this.props.selectedToken);
   }
 
-  componentWillReceiveProps(nextProps) {
```

Comment:
You moved this code to `componentDidUpdate`, and`componentDidUpdate` is not called in the initial render. Could this lead to bugs? Should we call this `getTokenSignature` piece of code also in the `componentDidMount`?

If yes, maybe we should create a method to wrap all the calls inside `componentDidMount` and `componentDidUpdate`

Reply:
I agree with the refactor, done in aad8472e8c7ac6edac7d81148f9276723bac2887

I also refactored it to reset the state when the user changes token and added a check to prevent a situation where the user changes tokens before the download completes causing the screen to show the administrative tab before the next request is complete

I also created an issue to refactor the mechanism for loading token data to cache locally instead of re-downloading every time the user changes token:

https://github.com/HathorNetwork/hathor-wallet/issues/356